sixtyfps_compiler/widgets/ugly/sixtyfps_widgets.60

```diff
@@ -308,6 +308,68 @@ export GroupBox := GridLayout {
     }
 }
 
+export TabWidgetImpl := Rectangle {
+    property <length> content_x: 0;
+    property <length> content_y: tabbar_preferred_height;
+    property <length> content_height: height - tabbar_preferred_height;
+    property <length> content_width: width;
+    property <length> tabbar_x: 0;
+    property <length> tabbar_y: 0;
+    property <length> tabbar_height: tabbar_preferred_height;
+    property <length> tabbar_width: width;
+
+    property <length> tabbar_preferred_height;
+    property <length> tabbar_preferred_width;
+    property <length> content_min_height;
+    property <length> content_min_width;
+    property <int> current_index;
+
+    border_width: 1px;
+    border_color: Palette.border_color;
+    border_radius: 2px;
+
+    preferred_width: content_min_width;
+    min_width: content_min_width;
+    preferred_height: content_min_height + tabbar_preferred_height;
+    min_height: content_min_height + tabbar_preferred_height;
+
+}
+export TabImpl := Rectangle {
+    property<string> title <=> t.text;
+    //property<image> icon;
+    property<bool> enabled : true;
+    property<bool> pressed;
+    property<int> current;
```

Comment:
This is like an "out" property, right - being two-way bound to the TabWidget's `current_index`?

Maybe the name could reflect that, like `current_index_ref` ?

(just a thought, feel free to ignore)

Reply:
it is a in/out property.

But it would be nice if the .60 language could document and enforce that.

Reply:
Right, so in the absence of that I think it makes sense to reflect that in the name. `current` is very bare-bones :-). Is it the current color, time or index? Maybe `tabwidget_current_index` to indicate that it is tied to the tab widget?